transaction/core/src/range.rs

```diff
@@ -29,17 +29,21 @@ impl Range {
 
     /// The number of indices in this Range.
     pub fn len(&self) -> u64 {
-        self.to - self.from + 1
+        if self.from <= self.to {
```

Comment:
Why did you opt for this versus returning a length of 0 when to > from?

Reply:
this allows us to support ranges like (-5, -2) which might be sensible for some future use

Reply:
Not really, from and to are u64.

Reply:
The main reason was, `len` is defined here to be "The number of indices in this Range", and so this sort of makes sense: the sequence 4,5,6 contains 3 elements, and the sequence 6,5,4 also contains 3 elements.

I'm open to alternatives, though. My main goals here are to make Range self-consistent and rule out possible panics, and elsewhere to make Range meaningful where it is used, like in Merkle proofs.